src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/Parser.cs

```diff
@@ -573,15 +582,10 @@ internal string ResolveVariables(Token inputToken, int offset, List<VariableRefe
                     offset: reference.Start - offset,
                     filename: inputToken.Filename);
                 string value = variableResolver.GetVariable(variablePos.Value, reference.VariableName);
-                if (value == null)
+                if (value == null || ThrowOnUnresolvedVariable == true /* Undefined variable */)
```

Comment:
the previous code was an `&&` conditional test.  is there a reason this should be an `||` now?

Reply:
The issue is "-variable" parameter functionality is not working as expected.

"ThrowOnUnresolvedVariable" is set to true only on ConfigureBatchParser() in ExecutionEngine.cs

When I pass powershell command "invoke-sqlcmd -query 'select $(a)' -Variable 'b=10'", ConfigureBatchParser() method is not called. So the "ThrowOnUnresolvedVariable" is not set and the '&&' condition fails.

Instead of getting error I am getting a result with empty data.

Reply:
it seems like the `ThrowOnUnresolvedVariable` flag no longer makes sense if we write the condition this way.  do we know why the code was originally written to only throw if both conditions are true?  i.e. in what conditions does value equal null and we don't want to throw.  clearly that was a case the original code author anticipated, inferred by the existence of `ThrowOnUnresolvedVariable`.  Basically I want to make sure we're not now throwing in situations that we "by design" didn't previously.

Reply:
This is no longer needed, we made the changes in Poweshell itself. So am closing this PR.